File: IT/1/Stringhe/Esercizi/Java/Esercizio14.java
/* file Esercizio14.java */

public class Esercizio14 {
	public static void main(String [] args) {
		String s = System.console().readLine("Scrivi una frase e leggi le singole parole: ");

		if (s.isEmpty()) {
			System.out.println("Stringa vuota!");
			return;
		}

		String [] parts = s.split("[\\s\\.,?!]+");

		for (String w : parts) {
			System.out.println(w);
		}
	}
}

File: IT/1/Stringhe/Esercizi/Python/esercizio14.py
# file esercizio14.py

from __future__ import print_function, unicode_literals
from sys import stdin, version_info
import re
if hasattr(__builtins__, 'raw_input'): input = raw_input

s = input("Scrivi una frase e leggi le singole parole: ")
if version_info[0] < 3: s = s.decode(stdin.encoding)

if not s:
	print("Stringa vuota!")
	exit()

parts = filter(None, re.compile(r"[\s\.,?!]+").split(s))

for w in parts:
	print(w)
